### ISSUE TITLE
Suppression Documentation is incomplete

### ISSUE THREAD

**Issue by marschwar** [detekt/detekt#6000]
## Expected Behavior
The [documentation about how to suppress findings](https://detekt.dev/docs/introduction/suppressing-rules) should mention all supported ways to do so.

## Observed Behavior
Currently only `RuleId` and `RuleSetId.RuleId` are mentioned. Missing are `all`, `detekt.all`, `detekt:all`,  `detekt.RuleSetId`, `detekt:RuleSetId`, `detekt.RuleSetId.RuleId` and  `detekt:RuleSetId:RuleId` .

Not sure if `all` really must be documented.

Relates to #5976

**Comment by atulgpt** [comment]
Why there are two delimiters `.` and `:`?

**Comment by cortinico** [comment]
I'm also unsure if we should support all of those formats

**Comment by marschwar** [comment]
I agree but this would be a breaking change. The colon separator seems to be the one used be other tools. I think we should keep that. I also vote to remove support for `all` and `detekt:all`.

We should still update the documentation to reflect the implementation and maybe mark those that will be removed in the future as deprecated.

**Comment by cortinico** [comment]
Yup it turns out both `:` and `.` are used by users:
https://github.com/search?type=code&auto_enroll=true&q=%40Suppress%28%22detekt

**Comment by marschwar** [comment]
Could this be a clean up we schedule for 2.x?
